packages/waku/src/lib/builder/build.ts

```diff
@@ -502,6 +508,14 @@ const emitHtmlFiles = async (
           typeof pathname === 'string' ? pathname2pathSpec(pathname) : pathname;
         let htmlStr = publicIndexHtml;
         let htmlHead = publicIndexHtmlHead;
+        if (cssAssets.length) {
+          const cssStr = cssAssets
+            .map((asset) => `<link rel="stylesheet" href="${asset}">`)
+            .join('\n');
+          // HACK is this too naive to inject style code?
```

Comment:
Yeah, I wish we could inject it in `rscIndexPlugin` for both server & client styles.
(hmm, we might consider about re-architecture at some point, for code-splitting styles, and (vite) plugins too.)

Reply:
Yea, we haveto do code-splitting at some point! Shall I now work on it and get it working with rscIndexPlugin?

Reply:
> Shall I now work on it and get it working with rscIndexPlugin?

Is it possible for the current architecture?
It would be nice if there's a less-hacky way.

Reply:
Sure I'll try my best, I'll get back to you soon! 
But if we want to do a release, I think let's merge it and tackle it later (maybe in the re-arch)

Reply:
(re-arch won't be soon.)
Yeah, if there's a better way, we would like to have it now. I plan to merge it next week.

Reply:
Took a look, the emitted css files are emitted in the client bundle phase which is the last phase, while rscIndexPlugin happens at the same time or even before in the buildSsrBundle phase. 

I don't know if it's possible to bring the client bundle before ssr! But it's not ideal anyway. Wdyt? maybe we should skip it.

Reply:
Yeah, it may require one more viteBuild step, which is not very good because it will make it slower.

I think the best is to send `<link>` tag in the RSC payload. But, it will require new technique (it's more work and different from re-arch I said above. It's more related with code-splitting.)

Thanks for your investigation. Let's merge this for now.